setup.py

```diff
@@ -1,71 +1,34 @@
 #!/usr/bin/python
 
-import sys
-import subprocess
-
-from setuptools import setup
-from setuptools import find_packages
-from setuptools.extension import Extension
+from setuptools import Extension, setup, find_packages
 
 __module__ = 'adbus'
 __url__ = 'https://github.com/ccxtechnologies'
 
 __version__ = None
 exec(open(f'{__module__}/__version__.py').read())
 
-if "--nosystemd" in sys.argv:
-    sys.argv.remove("--nosystemd")
-
-
-def cython(module, libraries):
-    if "--cythonize" in sys.argv:
-        sys.argv.remove("--cythonize")
-        from Cython.Build import cythonize
-        return cythonize(
-                [
-                        Extension(
-                                f"{__module__}.{module}",
-                                [f"{__module__}/{module}.pyx"],
-                                libraries=libraries
-                        )
-                ]
-        )
-    else:
-        return [
-                Extension(
-                        f"{__module__}.{module}", [f"{__module__}/{module}.c"],
-                        libraries=libraries
-                )
-        ]
-
-
-def check_external_dependancy(name):
```

Comment:
You should not check dependencies in that way. There are package / environment maintainers for that. For example, you don't check library version...Also you check for systemd, but not for libsystemd development files actually.